apps/src/codebridge/Console/index.tsx

```diff
@@ -33,6 +34,12 @@ const Console: React.FunctionComponent = () => {
     }
   }, [dispatch, levelId]);
 
+  useEffect(() => {
+    if (consoleRef.current) {
+      consoleRef.current.scrollTop = consoleRef.current.scrollHeight;
+    }
+  }, [consoleRef, codeOutput]);
+
```

Comment:
I don't think we need the `consoleRef` in the dependency list at all. Can it be dropped?

Otherwise, this is fine enough to approve, but just noting that this'll create a jumping scroll. So if the user has scrolled up to the top and runs it, it'll just hop straight to the bottom. Maybe that's okay, but maybe a smoother experience would be preferred.

To smooth it out, I think something along the following lines would work:
```
// note that we're not adding a ref to the whole container
<div className={moduleStyles.console}>
  { codeOutput.map() => { /* continue to map out console output as before */ }
  // add an extra placeholder at the end, which will now contain the scrollRef:
  <div ref={scrollAnchorRef} />
```

And change the effect to this:
```
  useEffect(() => {
    if (scrollAnchorRef.current) {
      scrollAnchorRef.current.scrollIntoView({ behavior: "smooth" });
    }
  }, [codeOutput]);
```

I _think_ I got all the logic right, but gimme a holler if it's not behaving properly and I'll help debug.

Of course, if we want to just keep the jumping behavior as is, then no need to change.

Reply:
Thanks! @ebeastlake also mentioned `scrollIntoView`, I updated it to follow this pattern! I updated the screen recording to the smooth scroll.